packages/store/src/usersSlice/thunks.ts

```diff
@@ -468,45 +562,47 @@ export const getPermissionSets = createAsyncThunk(`${sliceName}/getPermissionSet
               }, accu),
             { ...emptyUiPermissions, ...(permissionSetObject.result ?? {}) }
           );
-          const scopes = Object.values(scopedPermissionAreas).reduce((accu, { key, scopeType }) => {
+          const scopes = Object.values(scopedPermissionAreas).reduce((accu: UiPermissionsByAreaKeys[], { key, scopeType }) => {
             if (permissionSetObject.supported_scope_types?.includes(key) || permissionSetObject.supported_scope_types?.includes(scopeType)) {
               accu.push(key);
             }
             return accu;
           }, []);
           permissionSetObject = scopes.reduce((accu, scope) => {
-            accu.result[scope] = mapPermissionSet(permissionSetObject.name, [scopedPermissionAreas[scope].excessiveAccessSelector], scope);
+            accu.result[scope] = mapPermissionSet(permissionSetObject.name, [scopedPermissionAreas[scope].excessiveAccessSelector], scope) as any;
             return accu;
           }, permissionSetObject);
           accu[permissionSet.name] = permissionSetObject;
           return accu;
         },
         { ...getState().users.permissionSetsById }
       );
-      return Promise.all([dispatch(actions.receivedPermissionSets(permissionSets)), permissionSets]);
+      dispatch(actions.receivedPermissionSets(permissionSets));
+      return permissionSets;
```

Comment:
why?

Reply:
1. If possible the results of the thunk get assessed to check if e.g. thunk transforms get properly handled, since we often can't rely on the data as we get it. + having `thenable`s returned explicitly.
2. To not have the task dispatch happen only as a side effect but have the function be more explicit about what it does.

And unfortunately this change still is not in a corresponding commit, but swept up with the type additions (which again are what the task was about).